src/live_restore/live_restore_state.c

```diff
@@ -0,0 +1,444 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *  All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+#include "live_restore_private.h"
+
+/*
+ * __live_restore_state_to_string --
+ *     Convert a live restore state to its string representation.
+ */
+static void
+__live_restore_state_to_string(WTI_LIVE_RESTORE_STATE state, char *state_strp)
+{
+    switch (state) {
+    case WTI_LIVE_RESTORE_STATE_NONE:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_NONE");
+        break;
+    case WTI_LIVE_RESTORE_STATE_LOG_COPY:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_LOG_COPY");
+        break;
+    case WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION");
+        break;
+    case WTI_LIVE_RESTORE_STATE_CLEAN_UP:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_CLEAN_UP");
+        break;
+    case WTI_LIVE_RESTORE_STATE_COMPLETE:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_COMPLETE");
+        break;
+    }
+}
+
+/*
+ * __live_restore_state_from_string --
+ *     Convert a string to its live restore state.
+ */
+static int
+__live_restore_state_from_string(
+  WT_SESSION_IMPL *session, char *state_str, WTI_LIVE_RESTORE_STATE *statep)
+{
+
+    if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_NONE") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_NONE;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_LOG_COPY") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_LOG_COPY;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_CLEAN_UP") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_CLEAN_UP;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_COMPLETE") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_COMPLETE;
+    else
+        WT_RET_MSG(session, EINVAL, "Invalid state string: '%s' ", state_str);
+
+    return (0);
+}
+
+/*
+ * __live_restore_get_state_from_file --
+ *     Read the live restore state from the on-disk file. If it doesn't exist we return NONE. The
+ *     caller must already hold the live restore state lock. This function takes a *non-live
+ *     restore* file system, for example the backing posix file system used when accessing the
```

Comment:
Minor nit: "...non-live restore file system, for example, the backing posix file system, used..."  add two commas.